src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperDescriptorFactory.cs

```diff
@@ -255,10 +255,10 @@ private static TagHelperAttributeDescriptor ToAttributeDescriptor(PropertyInfo p
 
         private static bool IsAccessibleProperty(PropertyInfo property)
         {
-            return property.GetMethod != null &&
-                   property.GetMethod.IsPublic &&
-                   property.SetMethod != null &&
-                   property.SetMethod.IsPublic;
+            // Accessible properties are those with public getters and setters and without [HtmlAttributeNotBound].
+            return property.GetMethod?.IsPublic == true &&
+                property.SetMethod?.IsPublic == true &&
+                property.GetCustomAttribute<HtmlAttributeNotBoundAttribute>(inherit: false) == null;
```

Comment:
Don't you think it'd be odd if by doing:

https://github.com/aspnet/Razor/blob/de95f67400c8f20cf9cfe063c291720bf7d71cce/test/Microsoft.AspNet.Razor.Runtime.Test/TagHelpers/TagHelperDescriptorFactoryTest.cs#L1126-L1129

You end up with a different `HtmlAttributeName` association? I feel like it'd be misleading to see `public override string ValidAttribute1 { get; set; }` (no attribute) but you utilize it like `<inherited-overridden-attribute SomethingElse="..." />` with no attribute visible.

I don't feel very strongly either way sense inheriting from `TagHelper`s will be extremely rare; I just like the approach of clarity. Aka if you override you re-specify `HtmlAttributeName` to make it clear how to use it when re-scanning your own `TagHelper`s.